### PR TITLE
test (gradle-plugin) : Add Integration test to verify `jkube.debug.enabled` flag works as expected (#1214)

### LINKED ISSUE #1214: Gradle-Enricher: DebugEnricher
### Description

Child of #31

DebugEnricher works with Gradle

### AC
- [x] Enricher documentation is ported to Gradle
- [x] Create IT to verify that `jkube.debug.enabled` flag works as expected
- [x] kubernetes.yml is generated as expected
- [x] openshift.yml is generated as expected

### PR DESCRIPTION
**Pull request by rohanKanojia**
## Description
Fix #1214 #1271 

+ Add DebugModeIT for verifying `jkube.debug.enabled` property
+ Add documentation for DebugEnricher in Gradle Plugin Asciidocs

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [X] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [X] I have read the [contributing guidelines](https://www.eclipse.org/jkube/contributing)
 - [X] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have implemented unit tests to cover my changes
 - [X] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Comment by manusa** [comment]
I cannot find the debug enricher documentation for Maven. If there ain't, could you please create an issue so that we remember to add/port it?

File: gradle-plugin/it/src/it/debug-mode/build.gradle [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+                assembly {
+                    targetDir = '/deployments'
+                }
+                entryPoint {
+                    shell = "java -jar /deployments/${project.name}-${project.version}.jar"
+                }
+            }
+        }
+    }
+}
```
Comment (manusa) [inline gradle-plugin/it/src/it/debug-mode/build.gradle:49]: Any reason why we are using a custom image for this test?

Could we use something that, at least, relies on our base image that **does** have support for debug mode?

Comment (rohanKanojia) [reply gradle-plugin/it/src/it/debug-mode/build.gradle:49]: When using custom image, we don't need to provide extra files in `src/` directory

Comment (rohanKanojia) [reply gradle-plugin/it/src/it/debug-mode/build.gradle:49]: Are you asking to change `from` to some jkube base image? Bdw, this dummy image configuration is provided just to ensure resources are generated.

**Comment by rohanKanojia** [comment]
> I cannot find the debug enricher documentation for Maven. If there ain't, could you please create an issue so that we remember to add/port it?

I've created https://github.com/eclipse/jkube/issues/1271

**Review by sunix — APPROVED** [review]

**Review by manusa — APPROVED** [review]

**Pull request merged**